insights/parsers/lsscsi.py

```diff
@@ -0,0 +1,90 @@
+"""
+LsSCSI - command ``/usr/bin/lsscsi``
+====================================
+
+This parser reads the output of ``/usr/bin/lsscsi`` into a list of dictionarys.
+Each item is a dictionary with six keys:
+
+* ``HCTL`` - the scsi_host,channel,target_number,LUN tuple
+* ``Peripheral-Type`` - the SCSI peripheral type
+* ``Vendor`` - the vendor name
+* ``Model`` - the model name
+* ``Revision`` - the revision string
+* ``Primary-Device-Node`` - the primary device node name
+
+This dictionary is available in the ``data`` attribute.
+
+Parsing refers to http://sg.danny.cz/scsi/lsscsi.html.
+
+Sample input::
+
+    [1:0:0:0]    storage IET      Controller       0001  -
+    [1:0:0:1]    cd/dvd  QEMU     QEMU DVD-ROM     2.5+  /dev/sr0
+    [1:0:0:2]    disk    IET      VIRTUAL-DISK     0001  /dev/sdb
+    [3:0:5:0]    tape    HP       C5713A           H910  /dev/st0
+
+Examples:
+
+    >>> lsscsi[0]
+    {'Model': 'Controller', 'Vendor': 'IET', 'HCTL': '[1:0:0:0]', 'Peripheral-Type': 'storage', 'Primary-Device-Node': '-', 'Revision': '0001'}
+    >>> lsscsi.device_nodes
+    ['-', '/dev/sr0', '/dev/sdb', '/dev/st0']
+    >>> len(lsscsi.data)
+    4
+    >>> lsscsi[1]['Peripheral-Type']
+    'cd/dvd'
+"""
+
+from . import ParseException
+from .. import Parser, parser
+from insights.specs import Specs
+
+
+@parser(Specs.lsscsi)
+class LsSCSI(Parser):
+    """
+    Parse the output of ``/usr/bin/lsscsi``.
+    """
+    def parse_content(self, content):
+        if len(content) == 0:
+            # empty content of command output
+            self.data = []
+            return
+
+        LSSCSI_TABLE_HEADER_ITEMS = ['HCTL', 'Peripheral-Type', 'Vendor', 'Model', 'Revision', 'Primary-Device-Node']
+        LEN = len(LSSCSI_TABLE_HEADER_ITEMS)
+
+        col_index = None
+        pre_col_index = None
+        # Try to find the index of a proper six column line.
+        # If col_index can't be found in the above proper way,
+        # set it by squeezing redundant items to 'Model' column as a workaround.
+        for l in content:
```

Comment:
From the `loop`s logic and above comment, the `for`  loop seems unnecessary, you can get the `col_index` from the first line.
Of course, it's good to get the correct `col_index` when there is a line with ` len(col_split) == LEN `